sdk/core/azure-core/src/http/raw_response.cpp

```diff
@@ -15,17 +15,20 @@ HttpStatusCode RawResponse::GetStatusCode() const { return m_statusCode; }
 
 std::string const& RawResponse::GetReasonPhrase() const { return m_reasonPhrase; }
 
-std::map<std::string, std::string> const& RawResponse::GetHeaders() const { return this->m_headers; }
+std::map<std::string, std::string> const& RawResponse::GetHeaders() const
+{
+  return this->m_headers;
+}
 
-void RawResponse::AddHeader(uint8_t const* const begin, uint8_t const* const last)
+size_t RawResponse::AddHeader(uint8_t const* const begin, uint8_t const* const last)
 {
   // get name and value from header
   auto start = begin;
   auto end = std::find(start, last, ':');
 
   if (end == last)
   {
-    return; // not a valid header or end of headers symbol reached
+    return 1; // not a valid header or end of headers symbol reached
```

Comment:
Should we just throw in such cases? What action is the caller supposed to take if this fails?

Reply:
I would expect the caller to decide if they want to handle or throw.
If we throw, caller is forced to use try/catch to handle.
I want to let it be possible to ignore if the header is not added instead or signaling

Reply:
This is an invalid argument being passed in. What would the caller do to recover based on the return value of this operation?

Reply:
Don't want to throw. Caller can decide if they want to handle or just ignore.